### PR TITLE
Add webhook for automated deployment

### PR DESCRIPTION
**Pull request by denisfd**

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -24,7 +36,27 @@ GEM
...
+    rack (2.0.6)
+    rack-protection (2.0.4)
+      rack
     rake (12.3.1)
+    sinatra (2.0.4)
+      mustermann (~> 1.0)
+      rack (~> 2.0)
+      rack-protection (= 2.0.4)
+      tilt (~> 2.0)
+    sinatra-contrib (2.0.4)
```
Comment (mod) [inline Gemfile.lock]: is sinatra contrib absolutely necessary?

File: webhook.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+namespace '/deploy' do
+  before do
+    content_type 'application/json'
+  end
+
+  get '/ping' do
+    'pong'
+  end
+
+  post '/' do
```
Comment (mod) [inline webhook.rb]: get will be easier to call I believe
maybe /deploy/:payload

the payload being a JWT

File: webhook.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
+require 'sinatra'
```
Comment (mod) [inline webhook.rb]: your file must be in lib/microkube/server.rb
You can commit a Rackup file to have any web server running it
also include puma or thin in Gemfile to install a webserver

File: .env [hunk 4]
```diff
@@ -1,3 +1,5 @@
 COMPOSE_PROJECT_NAME=mikro
 APP_DOMAIN=app.local
 COMPOSE_FILE=compose/app.yaml:compose/backend.yaml:compose/gateway.yaml:compose/proxy.yaml:compose/daemons.yaml:compose/frontend.yaml:compose/utils.yaml
+PEATIO_IMAGE=rubykube/peatio:2.0.3-alpha
```
Comment (mod) [inline .env]: use only PEATIO_TAG
BARONG_TAG

File: compose/app.yaml [hunk 5]
```diff
@@ -2,7 +2,7 @@ version: '3.6'
 
 services:
   peatio:
-    image: rubykube/peatio:2.0.3-alpha
+    image: "${PEATIO_IMAGE_TAG}"
```
Comment (mod) [inline compose/app.yaml]: for now use  rubykube/peatio:${PEATIO_TAG}

File: lib/microkube/payload.rb [hunk 6]
```diff
@@ -0,0 +1,48 @@
+require 'jwt'
+
+class PayloadError < StandardError
```
Comment (mod) [inline lib/microkube/payload.rb]: move this inside Payload class
like 
Microkube::Payload::Error

File: lib/microkube/payload.rb [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+end
+
+module Microkube
+  class Payload
+    def initialize(params)
+      @jwt_secret = params[:secret]
+      @jwt_exp = params[:expire]
```
Comment (mod) [inline lib/microkube/payload.rb]: @expire = params[:expire] || 600

File: lib/microkube/payload.rb [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+      raise PayloadError.new unless @jwt_secret
```
Comment (mod) [inline lib/microkube/payload.rb]: validate @expire is > 0

File: lib/microkube/payload.rb [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+      @jwt_exp = 600 unless @jwt_exp
+    end
+
+    def encode(token)
+      JWT.encode token, @jwt_secret, 'HS256'
```
Comment (mod) [inline lib/microkube/payload.rb]: you don't need this method

File: lib/microkube/payload.rb [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+      raise PayloadError.new unless params[:tag]
+
+      token = params.merge({
+        'iat': Time.now.to_i,
+        'exp': (Time.now + @jwt_exp).to_i
+      })
+
+      encode token
+    end
+
```
Comment (mod) [inline lib/microkube/payload.rb]: name it verify!() if you don't need the other one don't implement it

File: lib/microkube/server.rb [hunk 11]
```diff
@@ -0,0 +1,51 @@
+require 'sinatra'
+require 'json'
+require_relative 'payload'
+
+services = %w[barong peatio frontend tower]
+
```
Comment (mod) [inline lib/microkube/server.rb]: later we are going to need to wrap this code in a class
Microkube::Server

File: lib/microkube/server.rb [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+before do
+  content_type 'application/json'
+end
+
+get '/deploy/ping' do
+  'pong'
+end
+
+get '/deploy/' do
```
Comment (mod) [inline lib/microkube/server.rb]: route must be /depoy/:token

File: lib/microkube/server.rb [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+  end
+
+  tag = "#{service.upcase}_IMAGE=#{image_tag}"
+  system "#{tag} docker-compose up -Vd #{service}"
+
+  answer(500, 'could not restart container') unless $?.success?
+  answer(200, "service #{service} updated with tag #{image_tag}")
+end
+
+def answer(status, message)
```
Comment (mod) [inline lib/microkube/server.rb]: use 

return answer()
instead of halt

File: lib/microkube/payload.rb (hunk 10 above)
Comment (denisfd) [reply lib/microkube/payload.rb]: Should I add method to verify params before merge?

**Pull request merged**